### PR TITLE
Add necessary changes to make format and lint workflows passing

### PR DESCRIPTION
**Pull request by mamueluth**
Kind of done. Currently, only fails because ament_cpplint does not find any files, which is true since all c++ files are located in templates/*. However, those files do not have a valid syntax at the moment and are therefore excluded. 

Besides this, the approach of using the ament_* commands has multiple drawbacks in my point of view.
- Need to make two "different" workflows:  
> - hard to comprehend for new users
> - need to be kept in sync 
- [ros-lint](https://github.com/ros-tooling/action-ros-lint) is not as flexible in excluding files as pre-commit, at least i wasn't able to exclude multiple files without listing them all
- Above-mentioned problem

### CODE REVIEW

File: .github/workflows/ci-lint.yml [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
+      - update_ci
 
 jobs:
   ament_lint:
     name: ament_${{ matrix.linter }}
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
+    container:
+      image: rostooling/setup-ros-docker:ubuntu-focal-ros-galactic-ros-base-latest
+      options: -u root
```
Comment (destogl) [inline .github/workflows/ci-lint.yml:15]: please remove.

File: .github/workflows/ci-lint.yml [hunk 2]
```diff
@@ -1,23 +1,35 @@
 name: Lint
 on:
+  workflow_dispatch:
   pull_request:
+  push:
+    branches:
+      - update_ci
```
Comment (destogl) [inline .github/workflows/ci-lint.yml:7]: please remove.

File: docs/conf.py [hunk 3]
```diff
@@ -244,6 +243,7 @@
     "REPOS_FILE_BRANCH": repos_file_branch,
 }
 
+
```
Comment (destogl) [inline docs/conf.py:246]:
```suggestion
```

Comment (mamueluth) [reply docs/conf.py]: Need to be 2 newlines because class definition follows

File: .github/workflows/ci-lint.yml (hunk 2 above)
Comment (mamueluth) [reply .github/workflows/ci-lint.yml]: done

**Pull request merged**